src/Filesystem/Filesystem.php

```diff
@@ -0,0 +1,257 @@
+        }
+
+        return new CallbackFilterIterator($directory, $filter);
+    }
+
+    /**
+     * Find files recursively in given directory path.
+     *
+     * @param string $path Directory path.
+     * @param mixed $filter If string will be used as regex for filtering using
+     *   `RegexIterator`, if callable will be as callback for `CallbackFilterIterator`.
+     *   Hidden directories (starting with dot e.g. .git) are always skipped.
+     * @param int|null $flags Flags for FilesystemIterator::__construct();
+     * @return \Traversable
+     */
+    public function findRecursive(string $path, $filter = null, ?int $flags = null): Traversable
+    {
+        $flags = $flags ?? FilesystemIterator::KEY_AS_PATHNAME
+            | FilesystemIterator::CURRENT_AS_FILEINFO
+            | FilesystemIterator::SKIP_DOTS;
+        $directory = new RecursiveDirectoryIterator($path, $flags);
+
+        $dirFilter = new RecursiveCallbackFilterIterator(
+            $directory,
+            function (SplFileInfo $current) {
+                if ($current->getFilename()[0] === '.' && $current->isDir()) {
+                    return false;
+                }
+
+                return true;
+            }
+        );
+
+        $flatten = new RecursiveIteratorIterator(
+            $dirFilter,
+            RecursiveIteratorIterator::CHILD_FIRST
+        );
+
+        if ($filter === null) {
+            return $flatten;
+        }
+
+        if (is_string($filter)) {
+            return new RegexIterator($flatten, $filter);
+        }
+
+        return new CallbackFilterIterator($flatten, $filter);
+    }
+
+    /**
+     * Dump contents to file.
+     *
+     * @param string $filename File path.
+     * @param string $content Content to dump.
+     * @return void
+     * @throws \Cake\Core\Exception\Exception When dumping fails.
+     */
+    public function dumpFile(string $filename, string $content): void
+    {
+        $dir = dirname($filename);
+        if (!is_dir($dir)) {
+            $this->mkdir($dir);
+        }
+
+        $exits = file_exists($filename);
+
+        if ($this->isStream($filename)) {
+            // @codingStandardsIgnoreLine
+            $success = @file_put_contents($filename, $content);
+        } else {
+            // @codingStandardsIgnoreLine
+            $success = @file_put_contents($filename, $content, LOCK_EX);
+        }
+
+        if (!$success) {
+            throw new Exception(sprintf('Failed dumping content to file "%s"', $dir));
+        }
+
+        if (!$exits) {
+            chmod($filename, 0666 & ~umask());
+        }
+    }
+
+    /**
+     * Create directory.
+     *
+     * @param string $dir Directory path.
+     * @param int $mode Octal mode passed to mkdir(). Defaults to 0755.
+     * @return void
+     * @throws \Cake\Core\Exception\Exception When directory creation fails.
+     */
+    public function mkdir(string $dir, int $mode = 0755): void
```

Comment:
Why use mkdir but then deleteDir copyDir?
what about either of these for consistency:

mkdir, rmdir, cpdir
mkDir, rmDir, cpDir
makeDir, removeDir, copyDir

Reply:
This class is only meant for internal use and won't be documented in the API hence there's no need to be too pedantic about the naming.